lib/dataset/sized_dict.cpp

```diff
@@ -364,6 +364,11 @@ bool SizedDict<Key, Value>::is_edges(const Key &key,
                         dim.has_value() ? *dim : val.dim());
 }
 
+template <class Key, class Value>
+void SizedDict<Key, Value>::set_aligned(const Key &key, const bool aligned) {
+  m_items.at(key).set_aligned(aligned);
```

Comment:
Do we need to check the readonly flag of the coords?

Reply:
Not sure. The flag is in `Variable`. So when slicing, we get a dict with new variables and setting the flag does not affect the original.
So I don't think anything breaks when setting the flag. But it can be surprising that it doesn't affect the original.

Reply:
We are checking the flag in `SizesDict::set` and `SizedDict::extract`, which would be used when turning a coord into an attr. By that logic we should check when we change the alignment flag.